application/F3DStarter.cxx

```diff
@@ -700,6 +708,123 @@ void F3DStarter::Render()
   f3d::log::debug("Render done");
 }
 
+//----------------------------------------------------------------------------
+void F3DStarter::SaveScreenshot(const std::string& filenameTemplate)
+{
+
+  const auto getScreenshotDir = []()
+  {
+    for (const char* const& candidate : { "XDG_PICTURES_DIR", "HOME", "USERPROFILE" })
```

Comment:
Any source to justify writing in these locations ?

Reply:
not really :(
I know for example Gnome saves screenshots to `~/Pictures/Screenshot` which would be equivalent to `$XDG_PICTURES_DIR/Screenshots`. `XDG_PICTURES_DIR` may not be really standard so `HOME` would be a safer fallback. `USERPROFILE`: I'm winging it hoping it would work on Windows

Reply:
- Windows: https://superuser.com/questions/1797103/where-did-my-screenshots-on-windows-10-get-stored-to
- MacOS: https://support.apple.com/en-us/102646
- Linux: https://wiki.archlinux.org/title/XDG_user_directories

Gnome seems to save in Screenshots, KDE directly in $XDG_PICTURES_DIR

There doesnt seems to be a standard. Lets do the best we can and document it thouroughly

Reply:
My concerns on this one are:
- the file should go somewhere that's easy to find for regular users (home, desktop, ...)
- I don't want to try and independently replicate OS behaviors (unless there's a straightforward way)

That's why so far it tries:
- `XDG_PICTURES_DIR` first, that would be something like `$HOME/Pictures` or whatever, localized as per user settings, looks like the best option
- otherwise `HOME` or `USERPROFILE` to still land in a user owned location regardless of OS
- last resort fallback to current directory, no idea where that would be depending on the OS and how the app is started. Hopefully it doesn't come to that in a regular setup

Open to other options if you can think of any

Reply:
I agree that there doesnt seem to be real standards and your approach is fine.

As always, just need good docs :)